packages/radio-group/src/vaadin-radio-group.js

```diff
@@ -73,510 +85,380 @@ class RadioGroupElement extends ThemableMixin(DirMixin(PolymerElement)) {
-          button.click();
-        } else {
-          button.checked = true;
-        }
-      } else {
-        button.checked = false;
-      }
-    });
-
-    this.validate();
-    this.readonly && this._updateDisableButtons();
-    button && this._setFocusable(this._radioButtons.indexOf(button));
+  /**
+   * @param {!CustomEvent} event
+   * @private
+   */
+  __onRadioButtonCheckedChange(event) {
+    if (event.target.checked) {
+      this.__selectRadioButton(event.target);
+    }
   }
 
-  /** @private */
-  _valueChanged(newV, oldV) {
-    if (oldV && (newV === '' || newV === null || newV === undefined)) {
-      this._changeSelectedButton(null);
+  /**
+   * Whenever the user sets a non-empty value,
+   * the method tries to select the radio button with that value
+   * showing a warning if no radio button was found with the given value.
+   * If the new value is empty, the method deselects the currently selected radio button.
+   * At last, the method toggles the `has-value` attribute considering the new value.
+   *
+   * @param {string | undefined} newValue
+   * @param {string | undefined} oldValue
+   * @private
+   */
+  __valueChanged(newValue, oldValue) {
+    if (!oldValue && !newValue) {
+      return;
+    }
+
+    if (oldValue && !newValue) {
+      this.__selectRadioButton(null);
       this.removeAttribute('has-value');
       return;
     }
 
-    if (!this._checkedButton || newV != this._checkedButton.value) {
-      const newCheckedButton = this._radioButtons.filter((button) => button.value == newV)[0];
+    if (this.__selectedRadioButton && this.__selectedRadioButton.value === newValue) {
+      return;
+    }
+
+    const newSelectedRadioButton = this.__radioButtons.find((radioButton) => {
+      return radioButton.value === newValue;
+    });
 
-      if (newCheckedButton) {
-        this._selectButton(newCheckedButton);
-        this.setAttribute('has-value', '');
-      } else {
-        console.warn(`No <vaadin-radio-button> with value ${newV} found.`);
-      }
+    if (newSelectedRadioButton) {
+      this.__selectRadioButton(newSelectedRadioButton);
+      this.toggleAttribute('has-value', true);
+    } else {
+      console.warn(`The radio button with the value "${newValue}" was not found.`);
     }
   }
 
   /**
-   * Returns true if `value` is valid.
+   * Whenever `readonly` property changes on the group element,
+   * the method updates the `disabled` property for the radio buttons.
    *
-   * @return {boolean} True if the value is valid.
+   * @param {!boolean} newValue
+   * @param {!boolean} oldValue
+   * @private
    */
-  validate() {
-    return !(this.invalid = !this.checkValidity());
+  __readonlyChanged(newValue, oldValue) {
+    if (newValue !== oldValue) {
+      this.__updateRadioButtonsDisabledProperty();
+    }
   }
 
   /**
-   * Returns true if the current input value satisfies all constraints (if any)
-   * @return {boolean}
+   * Extends the method from `DisabledMixin`
```

Comment:
Let's use the same writing as in other components:

```suggestion
   * Override method inherited from `DisabledMixin`
```

Reply:
Done.